### PR TITLE
Fix 7380: VPN easy access Region switcher

### LINKED ISSUE #7380: VPN easy access region switcher
### Description: 

VPN Easy Region Switcher

Details TBA

Search VPN ideas in Google Doc and read Easier Country Selection

Figma File VPN https://www.figma.com/file/PqLTp3DnANLQMM1XzaChqZ/Mobile-VPN?node-id=1662-21300&t=OZ2r2P1cvHZpvGcS-0

Section iOS: Country Selector Main Menu Addition

### Brave Version: 

- Can you reproduce this issue with the most recent build from TestFlight? 
- Can you reproduce this issue with the previous version of the current build from TestFlight? 
- Can you reproduce this issue with the current build from AppStore? 

### Website problems only:
- did you check with Brave Shields down? 
- did you check in Safari/Firefox (WkWebView-based browsers)?

### PR DESCRIPTION
**Pull request by soner-yuksel**
## Summary of Changes

This PR is adding easy access for region switcher in  menu.

The region switcher will fallow the rules of menu layout and it will be hidden when the VPN is off. When Automatic mode is selected the Title will represent current setting as automatic but the country flag will show the active region.

Other selection will follow the logic of showing country name and country flag.

Same with menu logic the region selector visual will follow same logic as when a website is loaded the region selection will not show subtitle but when an epoty tab is loaded the region selector will show the current title.

This pull request fixes #7380

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:

- Turn on / off VPN and check the region selector is being shown
- Change country or automatic region to see the selection is effected

## Screenshots:

https://user-images.githubusercontent.com/6643505/236286596-f419fe93-02c0-4229-9de4-2083efc8ac87.MP4

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/BraveVPN/BraveVPN.swift [hunk 1]
```diff
@@ -24,6 +24,10 @@ public class BraveVPN {
   /// This list is not static and should be refetched every now and then.
   static var regions: [GRDRegion] = []
   
+  /// Record last used region
+  /// It is used to hold details of the region when automatic slection is used
```
Comment (iccub) [inline Sources/BraveVPN/BraveVPN.swift]: `slection` typo

File: Sources/BraveVPN/BraveVPN.swift [hunk 2]
```diff
@@ -496,6 +512,32 @@ public class BraveVPN {
     }
   }
   
+  /// The function that fetched the last used region details from timezones
+  /// It used to get details of Region when Automatic Region is used
+  /// Otherwise the region detail items will be empty
+  /// - Parameter completion: comnpletion block that returns region with details or error
```
Comment (iccub) [inline Sources/BraveVPN/BraveVPN.swift]: `comnpletion` typo

File: Sources/BraveVPN/BraveVPN.swift (hunk 1 above)
Comment (soner-yuksel) [reply Sources/BraveVPN/BraveVPN.swift]: Fix at https://github.com/brave/brave-ios/pull/7391/commits/2853d0432e685ac5f48015c5d0304575bd698c51

File: Sources/BraveVPN/BraveVPN.swift (hunk 2 above)
Comment (soner-yuksel) [reply Sources/BraveVPN/BraveVPN.swift]: Fix at https://github.com/brave/brave-ios/pull/7391/commits/2853d0432e685ac5f48015c5d0304575bd698c51

**Review by iccub — APPROVED** [review]

**Pull request merged**